android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java

```diff
@@ -0,0 +1,354 @@
+    for (int i = 0; i < 4; i++) {
+      goldenResponses[i] = new Messages.StreamingOutputCallResponse();
+      goldenResponses[i].payload = new Messages.Payload();
+      goldenResponses[i].payload.type = Messages.COMPRESSABLE;
+    }
+    goldenResponses[0].payload.body = new byte[31415];
+    goldenResponses[1].payload.body = new byte[9];
+    goldenResponses[2].payload.body = new byte[2653];
+    goldenResponses[3].payload.body = new byte[58979];
+
+    final LinkedBlockingDeque<Messages.StreamingOutputCallResponse> responses =
+        new LinkedBlockingDeque();
+    final Messages.StreamingOutputCallResponse magicTailResponse =
+        new Messages.StreamingOutputCallResponse();
+    @SuppressWarnings("unchecked")
+    StreamObserver<Messages.StreamingOutputCallResponse> responseObserver =
+        new StreamObserver<Messages.StreamingOutputCallResponse>() {
+
+          @Override
+          public void onValue(Messages.StreamingOutputCallResponse value) {
+            responses.add(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+          }
+
+          @Override
+          public void onCompleted() {
+            responses.add(magicTailResponse);
+          }
+        };
+    StreamObserver<Messages.StreamingOutputCallRequest> requestObserver
+        = asyncStub.fullDuplexCall(responseObserver);
+    for (int i = 0; i < requests.length; i++) {
+      requestObserver.onValue(requests[i]);
+      assertEquals(goldenResponses[i], responses.poll(5, TimeUnit.SECONDS));
+      Assert.assertTrue("More than 1 responses received for ping pong test.", responses.isEmpty());
+    }
+    requestObserver.onCompleted();
+    Assert.assertEquals(magicTailResponse, responses.poll(5, TimeUnit.SECONDS));
+  }
+
+  public static void assertEquals(MessageNano expected, MessageNano actual) {
+    Assert.assertTrue("received message is not expected!",
+        MessageNano.messageNanoEquals(expected, actual));
+  }
+
+  private static void assertSuccess(StreamRecorder<?> recorder) {
+    if (recorder.getError() != null) {
+      throw new AssertionError(recorder.getError());
+    }
+  }
+
+  public static void assertEquals(List<? extends MessageNano> expected,
+      List<? extends MessageNano> actual) {
+    if (expected == null || actual == null) {
+      Assert.assertEquals(expected, actual);
+    } else if (expected.size() != actual.size()) {
+      Assert.assertEquals(expected, actual);
+    } else {
+      for (int i = 0; i < expected.size(); i++) {
+        assertEquals(expected.get(i), actual.get(i));
+      }
+    }
+  }
+
+  private SSLSocketFactory getSslSocketFactory(boolean useTestCa) throws Exception {
+    if (!useTestCa) {
+      return (SSLSocketFactory) SSLSocketFactory.getDefault();
+    }
+
+    SSLContext context = SSLContext.getInstance("TLS");
+    context.init(null, TMS , null);
+    return context.getSocketFactory();
+  }
+
+  public static void initTestCa(InputStream testCa) {
```

Comment:
Why is this static? It seems that this could just be passed to InteropTester's constructor.